runtime/onert/core/src/exporter/train/CheckpointExporter.cc

```diff
@@ -33,6 +33,52 @@ using namespace train;
 using namespace checkpoint;
 using namespace exec;
 
+class DataBuffer
+{
+public:
+  DataBuffer(uint32_t start_offset, uint32_t offset_size, uint32_t data_size)
+  {
+    _offset.resize(offset_size);
+    _data.resize(data_size);
+    _data_offset = start_offset + _offset.size() * sizeof(_offset[0]);
+
+    _data_ptr = _data.data();
+    _remaining_size = _data.size();
+  }
+
+  void setData(const char *data, uint32_t size)
+  {
+    assert(_index < _offset.size());
+    assert(size <= _remaining_size);
+
+    _offset[_index++] = _data_offset;
+    if (data && size > 0)
+      std::memcpy(_data_ptr, data, size);
+    _data_offset += size;
+    _data_ptr += size;
+
+    _remaining_size -= size;
+  }
+
+  void write(std::ofstream &ostream)
+  {
+    assert(_remaining_size == 0);
+    ostream.write(reinterpret_cast<const char *>(&_offset[0]), sizeof(_offset[0]) * _offset.size());
+    ostream.write(reinterpret_cast<const char *>(&_data[0]), _data.size());
+  }
+
+  uint32_t size() const { return sizeof(uint32_t) * _offset.size() + _data.size(); }
+
+private:
+  std::vector<uint32_t> _offset;
+  std::vector<char> _data;
```

Comment:
Once sizes of `_offset` and `_data` have been determined, they never grow up. 
Is there a reason to use `std::vector` instead of allocating heap memory? 

It could be my personal preference, 
but using heap allocation might simplify some code here.

Reply:
@zetwhite 

First, I think smart pointer is better idea than using raw pointer directly. :)
`_offset` should write `uint32_t` array data. However, if I use a smart pointer for this variable, I need to manage the pointer address to be write. It is more complicated than using `std::vector`. Even if using `std::vector` has a performance issue than a smart pointer, it would be more efficient to use the `std::vector` the term of implementing and managing, it would be more efficient to use the `std::vector` for implementation and management. What do you think about it?

Reply:
> First, I think smart pointer is better idea than using raw pointer directly. :)

Oh I see. I just thought using `raw_ptr` not a smart pointer. Definietly, using a smart pointer will not make the code simple. 

> What do you think about it? 

I agree. I'd prefer `vector` rather than smart pointer in this case :)